Add link guidance to use https instead of http (#1963)
Azure/azure-sdk: docs/policies/README-TEMPLATE.md

* **DO NOT** use version numbers in the links as the version numbers will often be missed when releasing new versions thus point to the wrong thing or not exist at all. Instead avoid linking directly to versioned information. If you still need a link you should first try to link to a page that contains all the versions so someone can choose or link to the latest which may be out of sync with the content but should usually remain valid.

> * **DO NOT** use `http` schema in the urls, please use seure link with `https` instead.

Reviewer: ```suggestion
* **DO** use `https` instead of `http` for links to ensure we are linking to secure servers.
```
